src/Npgsql/Internal/NpgsqlWriteBuffer.cs

```diff
@@ -127,17 +127,18 @@ public async Task Flush(bool async, CancellationToken cancellationToken = defaul
         } else if (WritePosition == 0)
             return;
 
-        var finalCt = cancellationToken;
-        if (async && Timeout > TimeSpan.Zero)
-            finalCt = _timeoutCts.Start(cancellationToken);
+        var finalCt = async && Timeout > TimeSpan.Zero
```

Comment:
Doesn't this mean that the user-provided cancellation token no longer works if there isn't a timeout?

Reply:
Yeah, you're right: there should have been just cancellationToken, instead of CancelllationToken.None.

Reply:
That may mean we're missing a test here (in case you feel like writing one)